internal/service/route53resolver/firewall_domain_list_data_source.go

```diff
@@ -0,0 +1,90 @@
+package route53resolver
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53resolver"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func DataSourceResolverFirewallDomainList() *schema.Resource {
```

Comment:
"Resolver" is inferred from the package name. Also, the datasource name should match the corresponding Resource. In this case ResourceFirewallDomainList.

```suggestion
func DataSourceFirewallDomainList() *schema.Resource {
```